plugins/samson_ledger/test/config/initializers/samson_ledger/client_test.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+require_relative "../../../test_helper"
+
+SingleCov.covered! file: 'plugins/samson_ledger/config/initializers/samson_ledger/client.rb'
+
+describe 'Ledger::Client' do
+  let(:deploy) { Deploy.first }
+
+  before do
+    stub_request(:post, "https://foo.bar/api/v1/events")
+  end
+
+  with_env(LEDGER_BASE_URL: 'https://foo.bar', LEDGER_TOKEN: "sometoken")
+
+  after do
+    ENV.delete("LEDGER_BASE_URL")
+    ENV.delete("LEDGER_TOKEN")
+  end
+
+  describe ".plugin_enabled?" do
+    it "is enabled" do
+      assert Ledger::Client.plugin_enabled?
+    end
+
+    it "is not enabled without token" do
+      ENV.delete("LEDGER_TOKEN")
+      refute Ledger::Client.plugin_enabled?
+    end
+
+    it "is not enabled without base_url" do
+      ENV["LEDGER_TOKEN"] = 'sometoken'
+      ENV.delete("LEDGER_BASE_URL")
+      refute Ledger::Client.plugin_enabled?
+    end
+  end
+
+  describe ".post_deployment" do
+    it "posts an event with a valid client" do
+      assert Ledger::Client.post_deployment(deploy)
```

Comment:
can do `@event_sent = stub_request` in the before and then `assert_requested @event_sent` here